core/coins/format.go

```diff
@@ -88,3 +88,74 @@ func FormatCoins(coins []*basev1beta1.Coin, metadata []*bankv1beta1.Metadata) (s
 
 	return strings.Join(formatted, ", "), nil
 }
+
+func ParseCoins(coins []string, metadata []*bankv1beta1.Metadata) ([]*basev1beta1.Coin, error) {
+	if len(coins) != len(metadata) {
+		return []*basev1beta1.Coin{}, fmt.Errorf("formatCoins expect one metadata for each coin; expected %d, got %d", len(coins), len(metadata))
+	}
+
+	parsedCoins := make([]*basev1beta1.Coin, len(coins), 0)
+	for i, coinStr := range coins {
+		coin, err := parseCoin(coinStr, metadata[i])
+		if err != nil {
+			return []*basev1beta1.Coin{}, err
+		}
+		parsedCoins = append(parsedCoins, coin)
+	}
+
+	return parsedCoins, nil
+}
+
+func parseCoin(coinStr string, metadata *bankv1beta1.Metadata) (*basev1beta1.Coin, error) {
+	coinArr := strings.Split(coinStr, " ")
+	coinDenom := coinArr[1]
```

Comment:
```suggestion
	coinAmt := coinArr[0]
	coinDenom := coinArr[1]
```

it makes reading the code below easier, instead of using `coinArr[0]` everywhere